### PR TITLE
request API

### PR DESCRIPTION
**Pull request by mcasper**
Opening this PR to test the waters and get feedback. If we like it, I'll go ahead and fully document everything/add the other HTTP methods.

Adds a `pub fn get<H: Handler>(path: &str, handler: H) -> Response` method for making requests to an Iron Handler easily in your tests. It's built such that each of the HTTP methods are just a thin layer over `make_request`, and the actual Request creation could potentially be extracted out to a macro, which would bring back the Constructor API originally outlined.

Will fulfill #21

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -8,9 +8,9 @@ repository = "https://github.com/reem/iron-test"
...
 
 [dependencies]
-iron = "0.2"
 hyper = "0.7"
-uuid = "0.1"
-url = "0.5"
+iron = "0.2"
 log = "0.3"
-
+router = "0.0.16"
```
Comment (reem) [inline Cargo.toml]: router should be set as a dev-dependency, and should be pinned to `0.0` instead of `0.0.16`

**Comment by reem** [comment]
This is a pretty good idea - I didn't think of having functions in this module take the Handler! I would say go ahead with this style of API, I think there is significant promise here to make testing much easier with a much more ergonomic API.

**Comment by mcasper** [comment]
Awesome! I'll go ahead and work on the rest of it

**Comment by mcasper** [comment]
@reem I think this is good to look over, although the docs could probably still use some love.

A note that urlencoded 0.1.2 looks like it doesn't build on stable due to a `#![feature(core, std_misc)]` annotation, so I had to point at the git url for now.

**Comment by mcasper** [comment]
@reem bump

File: README.md [hunk 2]
```diff
@@ -1,84 +1,66 @@
...
-let req = request::at_with(method::Put, Url::parse("http://www.google.com/").unwrap(), "Hello Google!");
-assert_eq!(req.method, method::Put);
-assert_eq!(req.url.serialize().as_slice(), "http://www.google.com/");
-assert_eq!(req.body.as_slice(), "Hello Google!");
+#[test]
+fn test_hello_world() {
+    let response = request::get("http://localhost:3000/hello", Headers::new(), HelloWorldHandler).unwrap();
+    let mut result_body = Vec::new();
+
+    {
```
Comment (reem) [inline README.md]: I wonder if we can make this part nicer too with some utility functions in iron-test

Comment (mcasper) [inline README.md]: I was wondering that too, I didn't know if it was out of the scope of this library or not. But anyone testing response bodies will have to do this, so it would definitely be a nice to have

Comment (reem) [inline README.md]: The scope of this library is basically "make testing iron apps easier" so
almost anything goes for now. Once we have more experience we will know
what works and doesn't and can narrow it down again.

Maybe an extension trait on Response for read_to_end or read_to_string
equivalents?

Comment (mcasper) [inline README.md]: 👍

Do we want to merge this PR in and I'll work on a followup utility functions branch, or do we want this to be all inclusive?

Comment (reem) [inline README.md]: Yeah, let's get this landed first and do it in another PR.

File: src/lib.rs [hunk 3]
```diff
@@ -66,45 +66,83 @@ pub mod mock {
...
+                                    path: &str,
+                                    body: &str,
+                                    mut headers: Headers,
+                                    handler: H) -> IronResult<Response> {
+            let data = Cursor::new(body.as_bytes().to_vec());
+            let mut stream = MockStream::new(data);
+            // Clone the stream so we can read the peer_addr off the original.
+            let mut stream_clone = stream.clone();
+            let mut reader = BufReader::new(&mut stream_clone as &mut NetworkStream);
+            let reader = HttpReader::EofReader(&mut reader);
```
Comment (reem) [inline src/lib.rs]: We can probably use a SizedReader here, since &str is sized.

Comment (reem) [inline src/lib.rs]: Read over the rest, everything looks good except this.

**Comment by mcasper** [comment]
Awesome, made the appropriate changes.

**Comment by reem** [comment]
Fantastic, thanks for this excellent rework of the interface.
